### ISSUE TITLE
[Bug]: No finite loss was found in best scalesgrid search

### ISSUE THREAD

**Issue by huypl53** [vllm-project/llm-compressor#2213]
### ⚙️ Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
### Environment Information ###
Operating System: `Linux-6.8.0-90-generic-x86_64-with-glibc2.39`
Python Version: `3.12.12 | packaged by conda-forge | (main, Oct 22 2025, 23:25:55) [GCC 14.3.0]`
llm-compressor Version: `0.9.0`
compressed-tensors Version: `0.13.0`
transformers Version: `4.57.3`
torch Version: `2.9.1+cu130`
CUDA Devices: `['NVIDIA RTX A6000']`
AMD Devices: `None`
```

</details>

### 🐛 Describe the bug

I tried to quantize "Qwen/Qwen3-VL-4B-Instruct" in AWQ then got error:
```
No finite loss was found in best scalesgrid search.
```

### 🛠️ Steps to reproduce

Here is my config:
```python
MODEL_ID = "Qwen/Qwen3-VL-4B-Instruct"  # Ensure this matches your model
SAVE_DIR = MODEL_ID.split("/")[-1] + "-AWQ-INT4"

# Load model in float16 (Safer for AWQ kernels than bfloat16)
print(f"Loading model {MODEL_ID}...")
model = AutoModelForImageTextToText.from_pretrained(
    MODEL_ID, dtype=torch.float16, device_map="auto", trust_remote_code=True
)
processor = AutoProcessor.from_pretrained(MODEL_ID, trust_remote_code=True)

# 2. CALIBRATION DATASET
# Using a small subset of neuralmagic/calibration for speed
DATASET_ID = "neuralmagic/calibration"
NUM_CALIBRATION_SAMPLES = 512
MAX_SEQUENCE_LENGTH = 4096

print("Loading dataset...")
ds = load_dataset(DATASET_ID, name="LLM", split=f"train[:{NUM_CALIBRATION_SAMPLES}]")
ds = ds.shuffle(seed=42)


def preprocess_function(example):
    messages = []
    for message in example["messages"]:
        messages.append(
            {
                "role": message["role"],
                "content": [{"type": "text", "text": message["content"]}],
            }
        )

    return processor.apply_chat_template(
        messages,
        return_tensors="pt",
        padding=False,
        truncation=True,
        max_length=MAX_SEQUENCE_LENGTH,
        tokenize=True,
        add_special_tokens=False,
        return_dict=True,
        add_generation_prompt=False,
    )


ds = ds.map(preprocess_function, batched=False, remove_columns=ds.column_names)


def data_collator(batch):
    # Filter out pixel_values if they are missing (text-only calibration)
    res = {}
    for key, value in batch[0].items():
        if key == "pixel_values":
            res[key] = torch.tensor(value, dtype=torch.float16).squeeze(0)
        else:
            res[key] = torch.tensor(value)
    return res
recipe = AWQModifier(
    mappings=[
        # AWQMapping(
        #     "re:.*input_layernorm$",
        #     ["re:.*q_proj$", "re:.*k_proj$", "re:.*v_proj$"],
        # ),
        AWQMapping("re:.*v_proj$", ["re:.*o_proj$"]),
        # AWQMapping(
        #     "re:.*post_attention_layernorm$",
        #     ["re:.*gate_proj$", "re:.*up_proj$"],
        # ),
        AWQMapping(
            "re:.*up_proj$",
            ["re:.*down_proj$"],
        ),
    ],
    ignore=[
        "re:.*embed_tokens",
        "re:.*input_layernorm$",
        "re:.*mlp[.]gate$",
        "re:.*post_attention_layernorm$",
        "re:.*norm$",
        "re:model[.]visual.*",
        "re:visual.*",
        "re:.*visual.*",
        "lm_head",
    ],
    duo_scaling=True,
    config_groups={
        "group_0": {
            "targets": ["Linear"],
            "weights": {
                "num_bits": 4,
                "type": "int",
                "symmetric": True,
                "group_size": 128,
                "strategy": "group",
                "dynamic": False,
                "actorder": None,
                "observer": "mse",
            },
        }
    },
)

# 4. RUN QUANTIZATION
print("Starting OneShot Quantization...")
oneshot(
    model=model,
    processor=processor,
    recipe=recipe,
    dataset=ds,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
    data_collator=data_collator,
)

# 5. SAVE
print(f"Saving model to {SAVE_DIR}...")
model.save_pretrained(SAVE_DIR, save_compressed=True)
processor.save_pretrained(SAVE_DIR)
print("Done!")
```

**Comment by huypl53** [comment]
I tried to reduce the calibration samples as `NUM_CALIBRATION_SAMPLES = 64` then it worked. Does this decrease the model performnace?

**Comment by brian-dellabetta** [comment]
Hi @huypl53 , if it works with fewer calibration samples, I wonder if one of the samples in the calibration set is causing it to NaN, or maybe the number is causing overflow? You can try with 128 or 256 samples to see if the issue persists or if recovery improves with more samples.

**Comment by huypl53** [comment]
> Hi [@huypl53](https://github.com/huypl53) , if it works with fewer calibration samples, I wonder if one of the samples in the calibration set is causing it to NaN, or maybe the number is causing overflow? You can try with 128 or 256 samples to see if the issue persists or if recovery improves with more samples.

thank for your response. I found the root problem. It had to be `bfloat16` (which is from qwen creator), not `float16` as mine. Here is the wrong code:
```python
# Load model in float16 (Safer for AWQ kernels than bfloat16)
print(f"Loading model {MODEL_ID}...")
model = AutoModelForImageTextToText.from_pretrained(
    MODEL_ID, dtype=torch.float16, device_map="auto", trust_remote_code=True
)
```
